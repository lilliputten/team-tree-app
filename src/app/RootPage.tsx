import { TPropsWithClassName } from '@/shared/types/generic';
import { siteConfig } from '@/config/site';
import { tailwindClippingLayout } from '@/lib/helpers/tailwind';
import { cn, constructMetadata } from '@/lib/utils';
import { ScrollArea } from '@/components/ui/scroll-area';
import { UseScrollableLayout } from '@/components/shared/ScrollableLayout';

export const pageTitle = 'Editor';
export const pageDescription = 'Editor page';

export const metadata = constructMetadata({
  title: pageTitle + ' - ' + siteConfig.name,
  description: pageDescription,
});

function DemoList(props: TPropsWithClassName) {
  const { className } = props;
  const itemsCount = 50;
  const items = Array.from(Array(itemsCount)).map((_none, no) => {
    const key = String(no);
    return <li key={key}>Item {no + 1}</li>;
  });
  return (
    <ScrollArea className={cn('__RootPage_DemoList', className, tailwindClippingLayout())}>
      <h3>List:</h3>
      <ul>{items}</ul>
    </ScrollArea>
  );
}

export function RootPage() {
  return (
    <div className={cn('__RootPage', tailwindClippingLayout())}>
      <UseScrollableLayout type="clippable" />
      <h1
        className={cn(
          'mb-4',
          'text-4xl',
          'font-heading',
          // 'font-extrabold',
          // 'leading-none',
          // 'tracking-tight',
          'text-gray-500',
          'dark:text-white',
          'md:text-5xl',
          'lg:text-6xl',
        )}
      >
        ПРОБА Welcome to <a href="https://nextjs.org">Next.js!</a>
      </h1>
      <DemoList />
    </div>
  );
}
