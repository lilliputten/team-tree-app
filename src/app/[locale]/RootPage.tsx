import { redirect } from 'next/navigation';
import { getTranslations } from 'next-intl/server';

import { getCurrentUser } from '@/lib/session';
import { constructMetadata } from '@/lib/utils';
import { dataRoute, welcomeRoute } from '@/constants/routes';
import { TAwaitedLocaleProps } from '@/i18n/types';

export async function generateMetadata({ params }: TAwaitedLocaleProps) {
  const { locale } = await params;
  const t = await getTranslations({ locale, namespace: 'RootPage' });
  return constructMetadata({
    title: t('title'),
    locale,
  });
}

export async function RootPage() {
  const user = await getCurrentUser();

  if (!user) {
    redirect(welcomeRoute);
  } else {
    redirect(dataRoute);
  }
}
