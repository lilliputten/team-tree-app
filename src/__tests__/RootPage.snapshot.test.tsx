/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';

import RootPage from '@/app/page';

it('RootPage snapshot', () => {
  const { container } = render(<RootPage />);
  expect(container).toMatchSnapshot();
});
