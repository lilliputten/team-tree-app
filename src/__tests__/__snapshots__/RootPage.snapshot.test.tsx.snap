// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootPage snapshot 1`] = `
<div>
  <div
    class="grid min-h-screen grid-rows-[20px_1fr_20px] items-center justify-items-center gap-16 p-8 pb-20 font-[family-name:var(--font-geist-sans)] sm:p-20"
  >
    <h1
      class="mb-4 text-4xl font-extrabold leading-none tracking-tight text-gray-500 dark:text-white md:text-5xl lg:text-6xl"
    >
      Welcome to 
      <a
        href="https://nextjs.org"
      >
        Next.js!
      </a>
    </h1>
  </div>
</div>
`;
